### PR TITLE
Restrucutre Pukki Postgresql documentation and add PostgreSQL 17

### PR DESCRIPTION
**Pull request by oscarkraemer**
## Proposed changes
This change restructure the Pukki PostgreSQL documentation to multiple section.
Add documentation describing differences between PostgreSQL 17 and 14.

## Checklist before requesting a review

- [ ] I have followed the instructions in the [Contributing](https://github.com/CSCfi/csc-user-guide/blob/master/CONTRIBUTING.md) and [Styleguide](https://github.com/CSCfi/csc-user-guide/blob/master/STYLEGUIDE.md) documents.
- [x] My pull request passes the automated tests. If not, visit [pull requests at Travis CI](https://app.travis-ci.com/github/CSCfi/csc-user-guide/pull_requests) and have a look at the job log.
- [x] I have included a link to the appropriate [preview page](https://csc-guide-preview.2.rahtiapp.fi/origin/DBAAS-334/cloud/dbaas/databases/) (select your branch from the list).

### CODE REVIEW

File: docs/cloud/dbaas/operations.md [hunk 1]
```diff
@@ -35,6 +35,26 @@ The commands to use:
 
 5. Verify with your preferred tool that your database is working as expected.
 
+## Major database upgrades
+Major version database upgrades works from the users point the same way but in the background
+happens more stuff, and is therefor more riskily.
```
Comment (lvarin) [inline docs/cloud/dbaas/operations.md:40]:
```suggestion
happens more operations, and is therefore riskier.
```

The sentence sounds still a bit clunky :/

File: docs/cloud/dbaas/postgres-accessing.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+## Command-line
+
+1. Install the `postgresql` command line tool. Note that some Linux distributions might provide ancient versions of it by default. Please consult [this page](https://www.postgresql.org/download/) for detailed installation instructions.
+2. Find the public IP of your database instance from its Overview tab in the web GUI, or with `openstack database instance list`.
+3. Use the following commands to access your PostgreSQL instance from the command line:
+   ```
+   psql --user ${USERNAME} --host ${PUBLIC_IP} ${DATABASE_NAME}
+   ```
+
+   The syntax normally used in configuration files is:
```
Comment (lvarin) [inline docs/cloud/dbaas/postgres-accessing.md:18]: which files? it should be clearer to mention the normal config file, or at least give some examples of names and locations.

File: docs/cloud/dbaas/operations.md [hunk 3]
```diff
@@ -35,6 +35,26 @@ The commands to use:
...
+
+Things that you need to take into account when doing a major database version upgrade:
+1. You have a recent enough backup that you can use if the upgrade fails.
+2. You have tested doing the upgrade on database restored from the backup.
+3. You reserve plenty of time for the upgrade.
+4. You have considered if you benefit of using a bigger flavor while doing the upgrade.
```
Comment (vartikashah) [inline docs/cloud/dbaas/operations.md:46]: You have considered if you would benefit from using a bigger flavor while doing the upgrade.

File: docs/cloud/dbaas/postgres-accessing.md (hunk 2 above)
Comment (oscarkraemer) [reply docs/cloud/dbaas/postgres-accessing.md]: I think I clarified the text.

File: docs/cloud/dbaas/operations.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (oscarkraemer) [reply docs/cloud/dbaas/operations.md]: Matti did rewrite the sentence.

File: docs/cloud/dbaas/operations.md [hunk 4]
```diff
@@ -35,6 +35,26 @@ The commands to use:
...
+5. You have ensured that your database have plenty of free disk space before starting the upgrade,
+you should aim for you volume be three times bigger than the used size.
```
[the remark this answers is not in the document]
Comment (oscarkraemer) [reply docs/cloud/dbaas/operations.md]: Matti rewrote the sentence.

File: docs/cloud/dbaas/postgres-extensions.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+ | unaccent                       | 1.1             | text search dictionary that removes accents |
+ | uuid-ossp                      | 1.1             | generate universally unique identifiers (UUIDs) |
+ | xml2                           | 1.1             | XPath querying and XSLT |
+
+
+## Parameters that users can modify
+
+The DBaaS allows users to modify some of the parameters.
+If there are some parameters that you think you should be able to modify, [please contact us](../../support/contact.md) and we'll see if we can make it possible.
+By default, we assume that default parameters are sane and that users should not, under normal circumstances, modify any of these parameters.
```
[the remark this answers is not in the document]
Comment (oscarkraemer) [reply docs/cloud/dbaas/postgres-extensions.md:88]: I think the word sane is what I wanted to use.

File: docs/cloud/dbaas/postgres-permissions.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+The command to grant permissions for a user to create tables to an existing database user:
+```sh
+openstack database user grant access $INSTNACE_ID $USER_NAME $DATABASE_NAME
+```
+
+The web interface also allows creating users and modifying their permissions.
+
+To create a new user with database access:
+    1. Choose an instance from the Instances page
```
[the remark this answers is not in the document]
Comment (oscarkraemer) [reply docs/cloud/dbaas/postgres-permissions.md]: Fixed.

File: docs/cloud/dbaas/postgres-permissions.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+### Giving a user read-only access to a table
+As the table owner or root user you can do the following SQL command:
+
+```sql
+GRANT SELECT ON ${table} TO ${user};
+```
+
+### Giving a user read-write access to a table
+
+If you want to give allow users to add, modify, remove and read rows in your database you can give the
```
[the remark this answers is not in the document]
Comment (oscarkraemer) [reply docs/cloud/dbaas/postgres-permissions.md]: Fixed.

File: docs/cloud/dbaas/postgres-permissions.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+You can always enable root access on a database instance and log in as the root user to modify
+permissions more freely.
+
+
+## A note about privileges
+
+If you have little or no prior experience with PostgreSQL, we recommend familiarizing yourself with how privileges in PSQL interact with Databases, Schema, and Tables. [Here's one tutorial that might be of use.](https://www.postgresqltutorial.com/postgresql-administration/postgresql-schema/)
+
+To avoid confusion, keep in mind that in PostgreSQL 14 the default privileges allow every user to connect to any database and create tables in the default 'public' schema. They cannot access existing tables or other schema without explicit permission, however, and they cannot create new schema.
```
[the remark this answers is not in the document]
Comment (oscarkraemer) [reply docs/cloud/dbaas/postgres-permissions.md]: Fixed.

File: docs/cloud/dbaas/postgres-versions.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+
+On PostgreSQL 14 database instances, the default permissions allow all database users to connect
+to any database and create tables in the public schema.
+
+On PostgreSQL 17 database instances, you need to explicitly grant users access to a database to
+allow them to create tables in the public schema of the specified database. This can be done as
+follows with the command line tools. This makes it a lot easier to create read-only users in
+PostgreSQL 17.
+
```
[the remark this answers is not in the document]
Comment (oscarkraemer) [reply docs/cloud/dbaas/postgres-versions.md]: The section was moved to postgresql-permissions.md and example added.

**Pull request merged**